src/components/TransactionHistoryRows/TransactionGenericRow.tsx

```diff
@@ -108,43 +106,16 @@ export const TransactionGenericRow = ({
   } = GetTxLayoutFormats({ parentWidth })
 
   const transfers = useMemo(() => {
-    return displayTransfers.map((transfer, index) => (
-      <Stack
-        alignItems='center'
-        key={index}
-        flex={1}
-        mt={{ base: 2, md: 0, xl: compactMode ? 2 : 0 }}
-        direction={index === 0 ? 'row' : 'row-reverse'}
-        textAlign={index === 0 ? 'left' : 'right'}
-      >
-        <AssetIcon
-          assetId={transfer.asset?.assetId}
-          boxSize={{ base: '24px', lg: compactMode ? '24px' : '40px' }}
-        />
-        <Box flex={1}>
-          <Amount.Crypto
-            color='inherit'
-            fontWeight='medium'
-            value={fromBaseUnit(transfer.value, transfer.asset?.precision ?? FALLBACK_PRECISION)}
-            symbol={transfer.asset?.symbol ?? FALLBACK_SYMBOL}
-            maximumFractionDigits={4}
-          />
-          {transfer.marketData.price && (
-            <Amount.Fiat
-              color='gray.500'
-              fontSize='sm'
-              lineHeight='1'
-              value={bnOrZero(
-                fromBaseUnit(transfer.value, transfer.asset?.precision ?? FALLBACK_PRECISION),
-              )
-                .times(transfer.marketData.price)
-                .toString()}
-            />
-          )}
-        </Box>
-      </Stack>
-    ))
-  }, [compactMode, displayTransfers])
+    return Object.entries(transfersByType).map(([, transfersOfType], index) => {
```

Comment:
big @kaladinlight : `could update Object.entries -> Object.values since we are just using the transfer arrays` - c6811ddbc1d9c53f64e8a613df93cd6ee57fe0bb